src/packet/mod.rs

```diff
@@ -10,6 +10,27 @@
 
 #![macro_use]
 
+pub enum PseudoHeaderIPv6IPv4 {
+    Ipv6PseudoHeader,
+    Ipv4PseudoHeader,
+}
+
+/// Used to help compute checksum value of packet
+pub trait PseudoHeader {
+    /// Converts IPv4 or IPv6 specific fields to a checksum value
```

Comment:
But how would we special case it? The pseudo header is used in the calculation of either the TCP or UDP checksum... a function that shouldn't know or care if IPv4 or IPv6 is used. hmm perhaps i've implemented the wrong api here. @little-dude approach seems pretty different and good.